### PR TITLE
Fixed so GUI is only manipulated in main thread

### PR DESCRIPTION
**Pull request by hgrehm**
What I thought had been random behavior caused by a race condition ended up being caused by manipulation of PyQt GUI in threads other than main. Checked threads calling any GUI-related functions to ensure only main does so and reconfigured those which didn't comply.

### CODE REVIEW

**Comment by hgrehm** [comment]
[b4ea649](https://github.com/UCSD-E4E/radio_collar_tracker_gcs/commit/b4ea6499921a31a6aefdc18b014386300cb1df74) resolves #42 
Component statuses must be manually updated if running simulator, ex: sim.setSystemState('STS_sysStatus', 3)

**Comment by hgrehm** [comment]
[a0d77e8](https://github.com/UCSD-E4E/radio_collar_tracker_gcs/pull/54/commits/a0d77e89093e3f89d636daf1430684e4fdc4a3f7) resolves #30
DataProvider feature had not been properly cleared + unimplemented function had been in exportPolygon function, both fixed

**Comment by ntlhui** [comment]
Bug:

Run Drone Simulator with the following `gcsConfig.ini`:
```
[FilePaths]
prefixpath = C:/Users/nthui/miniconda3/pkgs/qgis-3.28.2-py39hdacd352_1
prefixset = True

[LastCoords]
lat1 = 0.00017781252633007055
lat2 = 0.00017592349602529045
lon1 = -0.0007197364076617262
lon2 = -0.0007182240862076503

[Connection]
addr = 127.0.0.1
port = 9000
mode = drone
```
Result:
```
Exception has occurred: AttributeError
'RCTTCPServer' object has no attribute 'simList'
  File "D:\workspace\e4e\rct\radio_collar_tracker_gcs\scripts\droneSimulator.py", line 1006, in __init__
    while len(tsport.simList) == 0:
  File "D:\workspace\e4e\rct\radio_collar_tracker_gcs\scripts\droneSimulator.py", line 1079, in <module>
    sim = droneSimPack(args.port, args.target, args.protocol, args.clients);
AttributeError: 'RCTTCPServer' object has no attribute 'simList'
```

File: scripts/droneSimulator.py [hunk 1]
```diff
@@ -63,7 +62,7 @@
 from ping import rctPing
 
 
-def getIPs():
```
Comment (ntlhui) [inline scripts/droneSimulator.py:66]: Note: Unused

**Review by ntlhui — APPROVED** [review]

**Pull request merged**